diff --git a/lib/librte_hash/rte_cuckoo_hash.c b/lib/librte_hash/rte_cuckoo_hash.c
index 51db006..48c2a8b 100644
--- a/lib/librte_hash/rte_cuckoo_hash.c
+++ b/lib/librte_hash/rte_cuckoo_hash.c
@@ -380,6 +380,7 @@ rte_hash_hash(const struct rte_hash *h, const void *key)
 static inline hash_sig_t
 rte_hash_secondary_hash(const hash_sig_t primary_hash)
 {
+	return primary_hash + 1;
 	static const unsigned all_bits_shift = 12;
 	static const unsigned alt_bits_xor = 0x5bd1e995;
 
