diff --git a/lib/librte_hash/rte_cuckoo_hash.c b/lib/librte_hash/rte_cuckoo_hash.c
index 51db006..06b1a2c 100644
--- a/lib/librte_hash/rte_cuckoo_hash.c
+++ b/lib/librte_hash/rte_cuckoo_hash.c
@@ -449,6 +449,7 @@ make_space_bucket(const struct rte_hash *h, struct rte_hash_bucket *bkt)
 		next_bkt[i]->key_idx[j] = bkt->key_idx[i];
 		return i;
 	}
+	return -ENOSPC;
 
 	/* Pick entry that has not been pushed yet */
 	for (i = 0; i < RTE_HASH_BUCKET_ENTRIES; i++)
